cpp/data/frameProcessor/src/XspressProcessPlugin.cpp

```diff
@@ -345,6 +347,40 @@ void XspressProcessPlugin::setup_memory_allocation()
     free(inp_est_memblock_);
   }
   inp_est_memblock_ = malloc(sizeof(double) * this->frames_per_block_ * num_channels_);
+
+  if (sum_memblock_)
+  {
+    free(sum_memblock_);
+  }
+  sum_memblock_ = malloc(sizeof(double) * this->frames_per_block_);
+}
+
+
+/**
+ * Calculate the sum for a frame
+ *
+ * This method will receive a frame, and sum each element of data on it.
+ * The data is stored in the `sum_memblock_` array (allocated in the `setup_memory_allocation method`)
+ * and we iterate to by casting a pointer (`sum_ptr`) to the beginning of the array and moving it 
+ * by `num_scalars_recorded_` each time this method is called again.
+ * The 'num_scalars_recorded_' variable is set back to zero each time the frame arrays are flushed.
+ * i.e.: when num_scalars_recorded_ is 0 we move the sum_ptr pointer to index 0,
+ * when num_scalars_recorded_ is 1 we move the sum_ptr to index 1 based 
```

Comment:
I think my questions have made you overthink your explanation! I think this is all you need to say:

```suggestion
 * and each time this method is called it stores the result to this array at the offset `num_scalars_recorded_`, i.e.: when
 * `num_scalars_recorded_` is 0 we store the sum to index 0, when `num_scalars_recorded_` is 1 we store the sum to index 1.
 * The 'num_scalars_recorded_' variable is set back to zero each time the frame arrays are flushed.
```